goalc/compiler/Compiler.cpp

```diff
@@ -48,6 +52,18 @@ Compiler::Compiler(const std::string& user_profile, std::unique_ptr<ReplWrapper>
 
   // add GOOS forms that get info from the compiler
   setup_goos_forms();
+
+  m_nrepl_port = nrepl_port;
+  fmt::print("[nREPL]: Server Will Listen for a Connection on Port {}!\n\r", m_nrepl_port);
+  nrepl_thread = std::thread([&]() {
+    asio::io_context io_context;
+    ReplServer s(io_context, this);
```

Comment:
the tests are failing for a very strange reason (no asserts / no exceptions, just dies).  And ive narrowed it down to this line being the cause, the server doesn't even have to start running.

I assuming im doing something wrong with the pointer handling and such but,  not experienced enough to know the fix